fastasyncworldedit/API/fill-region-by-pattern.md

```diff
@@ -0,0 +1,54 @@
+# Filling a region using a pattern
+
+## Foreword
+First of all, patterns usually come from the core part of WorldEdit or FAWE. You can find all existing patterns [here](https://intellectualsites.github.io/fastasyncworldedit-javadocs/worldedit-core/com/sk89q/worldedit/function/pattern/Pattern.html)
+
+## Biome Cuboid Example
+Given are 3 things:
+- Position1: X=300, y=-64, z=300
+- Position2: X=600, y=128, z=600
+- BiomeType: Badlands
+
+```java
+public void setBiomeArea(World world, BlockVector3 position1, BlockVector3 position 2) {
+    // Here we must first adapt the world as in the WorldEdit example or translate it into the WorldEdit specialized object.
+    World faweWorld = BukkitAdapter.adapt(world);
+
+    //Now we define a square region using our 2 positions
+    CuboidRegion cuboidRegion = new CuboidRegion(position1, position2);
+
+    // Now we open an EditSession, which contains the complete process or information during an edit process in the world.
+    // Through the try-and-catch-with-resources we let Java automatically close the resource EditSession after use
+    try (EditSession editSession = WorldEdit.getInstance().newEditSession(faweWorld)) {
+        // Here we create our biome pattern
+        var pattern = new BiomeApplyingPattern(editSession, BiomeTypes.BADLANDS);
+
+        //This sets the pattern to the region
+        editSession.setBlocks((Region) cuboidRegion, pattern);
+    }
+}
+// This code would be called by the plugin in the main thread
+public void call() {
+    // Here we set our position 1 and 2
+    BlockVector3 position1 = BlockVector3.at(300,-64,300);
+    BlockVector3 position2 = BlockVector3.at(600,128,600);
```

Comment:
potentially follow the FAWE code style (spaces after commas separating parameters